orion/pkg/domains/user.go

```diff
@@ -0,0 +1,71 @@
+package domains
+
+import (
+	"database/sql"
+	"errors"
+	"github.com/ahsu1230/mathnavigatorSite/orion/pkg/sql_helper"
+	"regexp"
+	"time"
+)
+
+var TABLE_USERS = "users"
+
+type User struct {
+	Id         uint
+	CreatedAt  time.Time           `db:"created_at"`
+	UpdatedAt  time.Time           `db:"updated_at"`
+	DeletedAt  sql.NullTime        `db:"deleted_at"`
+	FirstName  string              `json:"firstName"`
+	LastName   string              `json:"lastName"`
+	MiddleName sql.NullString      `json:"middleName"`
```

Comment:
same thing as before.
Make `MiddleName` type string. Then in the Insert() and Upate() methods of the repo layer, give a sql.NullString instead of a string.